src/components/ReviewForm.tsx

```diff
@@ -67,206 +67,214 @@ const ReviewForm: any = (props: any) => {
+						<Select fullWidth {...field} error={Boolean(errors.semesterId)}>
+							<MenuItem value={'sp'}>Spring</MenuItem>
+							<MenuItem value={'sm'}>Summer</MenuItem>
+							<MenuItem value={'fa'}>Fall</MenuItem>
+						</Select>
+					)}
+					rules={{
+						validate: {
+							validateSemesterGivenYear: (semester) => {
+								clearErrors('year')
+								return validateSemesterYear(semester, getValues()['year'])
+							},
+							validateNotTakenCourse: (semester) => {
+								return validateUserNotTakenCourse(
+									userReviews,
+									courseData.courseId,
+									semester,
+									getValues()['year']
+								)
+							},
+						},
+					}}
+				></Controller>
+				{errors.semesterId &&
+					errors.semesterId.type === 'validateSemesterGivenYear' && (
+						<Alert severity='error'>{`Please wait until ${
+							fallbackDates[getValues()['semesterId']]
+						} to review ${courseData?.courseId} for semester ${
+							mapSemesterIdToName[`${getValues()['semesterId']}`]
+						} ${getValues()['year']}`}</Alert>
+					)}
+				{errors.semesterId &&
+					errors.semesterId.type === 'validateNotTakenCourse' && (
+						<Alert severity='error'>{`You've already reviewed this course for the semester and year!`}</Alert>
+					)}
+			</Grid>
+			<Grid item xs={12} lg={12}>
+				<InputLabel id='review-form-year'>Year</InputLabel>
+				<Controller
+					control={control}
+					name='year'
+					render={({ field }) => (
+						<Select {...field} fullWidth error={Boolean(errors.year)}>
+							{yearRange.map((year) => {
+								return (
+									<MenuItem key={year} value={year}>
+										<>{year}</>
+									</MenuItem>
+								)
+							})}
+						</Select>
+					)}
+					rules={{
+						validate: {
+							validateYearGivenSemester: (year) => {
+								clearErrors('semesterId')
+								return validateSemesterYear(getValues()['semesterId'], year)
+							},
+							validateNotTakenCourse: (year) => {
+								return validateUserNotTakenCourse(
+									userReviews,
+									courseData.courseId,
+									getValues()['semesterId'],
+									year
+								)
+							},
+						},
+					}}
+				></Controller>
+				{errors.year && errors.year.type === 'validateYearGivenSemester' && (
+					<Alert severity='error'>{`Please wait until ${
+						fallbackDates[getValues()['semesterId']]
+					} to review ${courseData?.courseId} for semester ${
+						mapSemesterIdToName[`${getValues()['semesterId']}`]
+					} ${getValues()['year']}`}</Alert>
+				)}
+				{errors.year && errors.year.type === 'validateNotTakenCourse' && (
+					<Alert severity='error'>
+						{`You've already reviewed this course for the semester and year!`}
+					</Alert>
+				)}
+			</Grid>
+			<Grid item xs={12} lg={12}>
+				<InputLabel id='review-form-workload'>Workload</InputLabel>
+				<Controller
+					control={control}
+					name='workload'
+					render={({ field }) => (
+						<TextField
+							{...field}
+							defaultValue={null}
+							type='number'
```

Comment:
If possible to define this around something like `TRating | 0`, that would be better...but if not, let's not split hairs over that for now